src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -114,13 +92,44 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
   }
 
   /**
+   * Gets the current account.
+   *
+   * @return AccountInterface
+   */
+  public function getAccount() {
+    if (empty($this->account)) {
+      $this->setAccount(\Drupal::currentUser()->getAccount());
+    }
+
+    return $this->currentUser;
+  }
+
+  /**
+   * Sets the current object account.
+   *
+   * @param AccountInterface $account
+   */
+  public function setAccount(AccountInterface $account) {
+    $this->currentUser = $account;
+  }
+
+  /**
    * Get the user's groups.
    *
    * @return ContentEntityInterface[]
    */
-  private function getUserGroups() {
+  protected function getUserGroups() {
     $other_groups = Og::getEntityGroups('user', $this->getAccount()->id());
     return $other_groups[$this->configuration['target_type']];
   }
 
+  /**
+   * Determines if the current has hs group admin permission.
+   *
+   * @return bool
+   */
+  protected function isGroupAdmin() {
+    return $this->getAccount()->hasPermission('administer group');
```

Comment:
Related to comment where this is used above. I think this is all that was used before to determine this?